### PR TITLE
Rewrite & update to page-contracts

### LINKED ISSUE #2604: Specify gas costs in mega gas
There are upcoming changes for the contracts pallet which effectively unify gas and weight (https://github.com/paritytech/substrate/pull/5712). What this means for contract developers is that the gas costs are _way_ higher now as they are specified in weight (we still call it gas for now when communicating with contract developers).

Previously the cost for one wasm instruction was 1 gas. Now it is 500_000 weight = gas where every weight represent 1 pico second of execution on our reference machine. Currently, the contract UI only allows for 9 numbers to be entered which does not allow for any meaningful calculation.

I propose to change the field so that you specify **mega** gas which means that we have a granularity of micro second execution time there. Maybe even add a dynamically displayed calculation that shows how much execution time of the whole block the entered gas limit constitutes with a nice progress bar.

Example:
We enter 1_000_000 mega gas. Then we could display something like this:
1 second execution time, 50% of Block, [Progress Bar]

### PR DESCRIPTION
**Pull request by kwingram25**
Closes #2604 

- Migrate classes to functional
- Rewrite Row and related
- Extra hooks

### CODE REVIEW

File: packages/page-contracts/src/Contracts/Call.tsx [hunk 1]
```diff
@@ -35,10 +36,10 @@ function Call (props: Props): React.ReactElement<Props> | null {
...
   const callMessage = callContract?.getMessage(isNull(callMessageIndex) ? undefined : callMessageIndex);
 
-  const [accountId, setAccountId] = useState<StringOrNull>(null);
-  const [endowment, setEndowment] = useState<BN>(new BN(0));
-  const [gasLimit, setGasLimit] = useState<BN>(new BN(DEFAULT_GAS_LIMIT));
-  const [isBusy, setIsBusy] = useState(false);
+  const [accountId, setAccountId] = useAccountId();
+  const [endowment, isEndowmentValid, setEndowment] = useFormField<BN>(new BN(0));
+  const [gasLimit, isGasLimitValid, setGasLimit] = useNonZeroBn(new BN(GAS_LIMIT));
+  const [isBusy, , setIsBusy] = useToggle();
```
Comment (erler) [inline packages/page-contracts/src/Contracts/Call.tsx:42]:
```suggestion
  const [isBusy, setIsBusy] = useToggle();
```

Comment (jacogr) [reply packages/page-contracts/src/Contracts/Call.tsx:42]: This is not quite correct - `[isBusy, toggleIsBusy, setIsBusy]`, so the usage here is correct.

Comment (erler) [reply packages/page-contracts/src/Contracts/Call.tsx:42]: Ah sorry, my bad

Comment (jacogr) [reply packages/page-contracts/src/Contracts/Call.tsx:42]: No issues, extra eyeballs are _always_ good.

File: packages/react-components/src/Tags.tsx [hunk 2]
```diff
@@ -46,6 +46,7 @@ function Tags ({ children, className, color = 'grey', isEditable, isEditing, onC
 
   const _onSave = useCallback(
     (): void => {
+      console.log('_onSave');
```
Comment (jacogr) [inline packages/react-components/src/Tags.tsx:49]:
```suggestion
```

**Pull request merged**